content/docs/use-cases/ci-and-ml.md

```diff
@@ -0,0 +1,171 @@
+# CI for ML
+
+## Motivation
+
+ML engineers and data scientists face many unique challenges. Some of these
+challenges can be solved by adopting first-citizen concepts from traditional
+software engineering and development. Probably the most common adoption is
+versioning control. Some other, unfortunately less common, could be unit testing
+data processing steps of a pipeline. Here is a non exhaustive list of a few
+other unique challenges:
+
+- Automatically re-train models upon changes.
+- Compare (and report) performance of different versions of a model.
+- Test data correctness when new data is introduced.
+- Handle a model training that is computationally intensive.
```

Comment:
Minor: these aren't complete sentences so no need for period `.`, I think.

Reply:
Based on [rule #1 here](https://writingcooperative.com/when-and-when-not-to-insert-punctuation-at-the-end-of-a-bullet-point-cefd3915950d) I would say that the punctuation is correct. But maybe I misunderstand it.

Reply:
I can't open that because of Mediums article reading limits... Can you paste the rule? According to this one they shouldn't have periods AFAIU.

We do have guidelines around this in https://dvc.org/doc/user-guide/contributing/docs#doc-style-guidelines-javascript-and-markdown, please check out that doc if you haven't.

Reply:
(this one = https://www.free-power-point-templates.com/articles/rules-for-using-periods-with-bullet-points/)

Reply:
If it was a blog post I wouldn't care at all but we do have lots of other bullet lists in existing docs so the main thing here is to keep consistency with existing docs. Or if other docs are wrong, to open an issue or small PR to fix them 🙂

Reply:
Thanks for the pointer!

I really don't know. I feel comfortable with capitalization and punctuation as it is now. I'm also fine with changing it.

Here is the text from the [link](https://writingcooperative.com/when-and-when-not-to-insert-punctuation-at-the-end-of-a-bullet-point-cefd3915950d):

> Use a period at the end of a bullet point when the bullet point:
Completes a sentence based on its stem.
Is a complete sentence on its own.
The “stem” is made up of the words that end with a colon in the introduction to the bullet points. In the above statement, “Use a period at the end of a bullet point when the bullet point:” is the stem. Since each bullet point in the series that follows the stem completes the sentence begun by the stem, each ends with a period.

BTW and off-topic: in https://github.com/iterative/dvc.org/blame/master/content/docs/user-guide/contributing/docs.md#L185-L187. I think that `commit` is git oriented while `revision` and `reference` are maybe more SVN? I would say that the most neutral term is `version`.... But as I said. off-topic.

Reply:
Thanks (I was able to open it with an incognito windows too).

These bullet items don't complete the sentence. It would need to be something like:

- this. `  # It would need to be something like this.` ✅ 
- or that. `  # It would need to be something like that.` ✅ 

But not when you say one of the following:

- one `  # But not when you say one of the following one?` ❌ 
- the other `  # But not when you say one of the following the other?` ❌ 

The difference is subtle and this is definitely too deep haha but yeah if you don't mind please don't use periods here.

Reply:
> commit is git oriented while revision and reference are maybe more SVN? I would say that the most neutral term is version

Maybe all those terms are both in Git and other SCM tools. Commit being the more basic one but indeed "version" is even more general. We use commit extensively throughout docs since we expect 99% of our users know what that is. Maybe that's a wrong assumption, I'm not sure TBH. So far no complaints 🙂